twr_ke18f: tests/subsys/pm/power_mgmt_soc: build failure

**Describe the bug**

tests/subsys/pm/power_mgmt_soc build failure

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using?
 twr_ke18f
 - What have you tried to diagnose or workaround this issue?
 - Is this a regression? If yes, have you been able to "git bisect" it to a
   specific commit?

git bisect find this commit introduce issue
```
b84b6de76cc soc: nxp: kinetis: add soc support for mke17z7
```
 - ...

**To Reproduce**
Steps to reproduce the behavior:



**Expected behavior**
build PASS

**Impact**
low power application

**Logs and console output**

```
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/timer/mcux_lptmr_timer.c
In file included from /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/toolchain.h:50,
                 from /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/sys/util.h:18,
                 from /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/init.h:13,
                 from /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/timer/mcux_lptmr_timer.c:9:
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/toolchain/gcc.h:87:36: error: static assertion failed: "No LPTMR instance enabled in devicetree"
   87 | #define BUILD_ASSERT(EXPR, MSG...) _Static_assert(EXPR, "" MSG)
      |                                    ^~~~~~~~~~~~~~
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/timer/mcux_lptmr_timer.c:16:1: note: in expansion of macro 'BUILD_ASSERT'
```

**Environment (please complete the following information):**
 - OS: (e.g. Linux, MacOS, Windows)
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA or Version used: fe03e0c4bb3f8a658efe69b04c430d1a15474c6c

--- a/drivers/counter/Kconfig.mcux_lptmr
+++ b/drivers/counter/Kconfig.mcux_lptmr
@@ -12,7 +12,7 @@ config COUNTER_MCUX_LPTMR
 	  Enable support for the MCUX Low Power Timer (LPTMR).
 
 config COUNTER_MCUX_KINETIS_LPTMR
-	bool "Deprecated DT compatible"
+	bool
 	default y
 	depends on DT_HAS_NXP_KINETIS_LPTMR_ENABLED
 	select DEPRECATED

--- a/drivers/timer/Kconfig.mcux_lptmr
+++ b/drivers/timer/Kconfig.mcux_lptmr
@@ -6,7 +6,8 @@
 config MCUX_LPTMR_TIMER
 	bool "MCUX LPTMR timer"
 	default y
-	depends on DT_HAS_NXP_KINETIS_LPTMR_ENABLED
+	depends on DT_HAS_NXP_KINETIS_LPTMR_ENABLED || \
+		   MCUX_KINETIS_LPTMR
 	depends on !COUNTER_MCUX_LPTMR
 	depends on PM
 	select SYSTEM_TIMER_HAS_DISABLE_SUPPORT
@@ -14,3 +15,13 @@ config MCUX_LPTMR_TIMER
 	  This module implements a kernel device driver for the NXP MCUX Low
 	  Power Timer (LPTMR) and provides the standard "system clock driver"
 	  interfaces.
+
+config MCUX_KINETIS_LPTMR
+	bool
+	default y
+	depends on DT_HAS_NXP_KINETIS_LPTMR_ENABLED
+	select DEPRECATED
+	help
+	  The compatible string "nxp,kinetis-lptmr" should
+	  be swiched to "nxp,lptmr" in DT. The former will
+	  be removed eventually.

--- a/drivers/timer/mcux_lptmr_timer.c
+++ b/drivers/timer/mcux_lptmr_timer.c
@@ -4,7 +4,12 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#include <zephyr/devicetree.h>
+#if DT_HAS_COMPAT_STATUS_OKAY(nxp_kinetis_lptmr)
 #define DT_DRV_COMPAT nxp_kinetis_lptmr
+#else
+#define DT_DRV_COMPAT nxp_lptmr
+#endif
 
 #include <zephyr/init.h>
 #include <zephyr/drivers/timer/system_timer.h>

